src/waveform/renderers/waveformmark.cpp

```diff
@@ -335,7 +335,7 @@ class MarkerGeometry {
 };
 
 QImage WaveformMark::generateImage(float devicePixelRatio) {
-    assert(needsImageUpdate());
+    DEBUG_ASSERT(needsImageUpdate());
```

Comment:
VERIFY_OR_DEBUG_ASSERT ?
IIUC all below would be a no-op otherwise, right?

Reply:
Yes, but we assert that it never happens and when it happens, it does no harm.